### PR TITLE
SP017. User search using Ransack gem

### PR DESCRIPTION
**Pull request by HumRuslan**
**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [ ] "let!" is used in specs
- [x] screenshot of feature is added if task has UI

### CODE REVIEW

**Comment by HumRuslan** [comment]
users

File: app/models/user.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@ class User < ApplicationRecord
   before_create :set_default_role
 
   scope :confirmed, -> { where.not(confirmed_at: nil) }
-
+  ransack_alias :user, :first_name_or_last_name
```
Comment (chu4) [inline app/models/user.rb:16]: pls add blank lines before and after the line

File: app/controllers/account/admin/user_controller.rb [hunk 2]
```diff
@@ -1,7 +1,8 @@
 class Account::Admin::UserController < Account::Admin::AdminApplicationController
   before_action :find_user, only: %i[blocked activated add_admin remove_admin]
   def index
-    @users = User.user.confirmed
+    @q = User.ransack(params[:q])
+    @users = @q.result(distinct: false).user.confirmed
     @admins = User.admin
```
Comment (chu4) [inline app/controllers/account/admin/user_controller.rb:6]: what about search by admins?

File: app/controllers/account/admin/user_controller.rb [hunk 3]
```diff
@@ -26,6 +27,11 @@ def remove_admin
     redirect_to account_admin_user_index_url
   end
 
+  def search
```
Comment (chu4) [inline app/controllers/account/admin/user_controller.rb]: whats' the role of this one? why not just use index?

**Review by chu4 — APPROVED** [review]

**Pull request merged**